daemon/indexing/listener/listener.js

```diff
@@ -182,9 +174,40 @@ async function runBatch(opts, context) {
   return lastLogBlock
 }
 
+async function withRetrys(fn) {
+  let tryCount = 0
+  while (true) {
+    try {
+      await fn() // Do our action.
+      return // it worked!
```

Comment:
should this return whatever the fn returned ? just for future flexibility.

Reply:
Yeah, that would be nice. Good thinking.